File: apps/backend-java/src/main/java/com/housingplatform/admin/dto/HousingRequestDto.java
package com.housingplatform.admin.dto;

import com.housingplatform.persistence.enums.AccommodationType;
import com.housingplatform.persistence.enums.HousingRequestStatus;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

public record HousingRequestDto(
    UUID id,
    String email,
    String desiredArea,
    LocalDate checkIn,
    LocalDate checkOut,
    Integer budgetMax,
    AccommodationType accommodationType,
    String notes,
    HousingRequestStatus status,
    OffsetDateTime createdAt) {}

File: apps/backend-node/src/features/admin/dto/housing-request.dto.ts
import type { HousingRequestStatus } from './housing-request-status.dto';

export interface HousingRequestDto {
  id: string;
  email: string;
  desiredArea: string;
  checkIn: string | null;
  checkOut: string | null;
  budgetMax: number | null;
  accommodationType: string | null;
  notes: string | null;
  status: HousingRequestStatus;
  createdAt: string;
}

export interface UpdateHousingRequestStatusDto {
  status: HousingRequestStatus;
}
